components/Label/Label.test.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { Label } from './Label'; 
+
+describe('Label', () => {
+  const disabledVariants = [true, false]; 
+
+  disabledVariants.forEach((disabled) => {
+    it(`renders correctly when disabled is ${disabled}`, () => {
+      render(
+        <Label data-testid="test-label" disabled={disabled}>
+          Test Label
+        </Label>
+      );
+
+      const label = screen.getByTestId('test-label');
+      expect(label).toBeInTheDocument();
+
+      const expectedClass = disabled ? 'opacity-50 cursor-not-allowed' : 'peer-aria-checked:border-accent peer-hover:border-white';
+      expect(label).toHaveClass(expectedClass);
+    });
+  });
+
+  it('applies additional custom classes correctly', () => {
+    const extraClasses = 'custom-class-label extra-custom-class';
+    render(
+      <Label
+        data-testid="custom-class-label"
+        className={extraClasses}
+      >
+        Custom Test Label
+      </Label>
+    );
+
+    const labelCustom = screen.getByTestId('custom-class-label');
+    expect(labelCustom).toBeInTheDocument();
+    expect(labelCustom).toHaveClass(extraClasses);
+  });
+});
```

Comment:
Q: I'm not sure I understand the purpose of this test. I don't see these specific custom classes in your Label component, could you explain your thought process behind this please?

Reply:
The Label component has the cva classes, but it also accepts a className prop to allow for custom styles. This test checks that any additional classes are passed and correctly applied together with the variant-based classes.